src/components/network/layers/fork-line-layer.js

```diff
@@ -32,21 +39,34 @@ export default class ForkLineLayer extends LineLayer {
             'vs:#decl': `
 attribute float instanceLineParallelIndex;
 attribute float instanceLineAngle;
+attribute float instanceOffsetStart;
+attribute float instanceProximityFactor;
 uniform float distanceBetweenLines;
 uniform float maxParallelOffset;
 uniform float minParallelOffset;
+uniform float substationRadius;
+uniform float substationMaxPixel;
+uniform float minSubstationRadiusPixel;
             `,
             'float segmentIndex = positions.x': `;
-target = source ;
-if( abs(instanceLineParallelIndex) != 9999. ) {
+    target = source ;
     float offsetPixels = clamp(project_size_to_pixel( distanceBetweenLines), minParallelOffset, maxParallelOffset );
     float offsetCommonSpace = project_pixel_size(offsetPixels);
-    vec4 trans = vec4(cos(instanceLineAngle), -sin(instanceLineAngle ), 0, 0.) * instanceLineParallelIndex ;
-    trans.x -= sin(instanceLineAngle) ;
-    trans.y -= cos(instanceLineAngle) ;
-    trans = trans * offsetCommonSpace;
-    target+=project_common_position_to_clipspace(trans) - project_uCenter;
-}
+
+    float offsetSubstation = clamp(project_size_to_pixel(substationRadius*instanceOffsetStart ), 
+                                    minSubstationRadiusPixel, 
+                                    substationMaxPixel * instanceOffsetStart );
+    float offsetSubstationCommonSpace = project_pixel_size(offsetSubstation) ;
+
+    vec4 trans = vec4(cos(instanceLineAngle), -sin(instanceLineAngle ), 0, 0.) * instanceLineParallelIndex;
+
+    if( instanceLineParallelIndex != 0. ){
```

Comment:
Why do we need this special case ? Doesn't it lead to a mismatch with the arrow layer ?